tests/deepspeed/test_deepspeed_multiple_model.py

```diff
@@ -0,0 +1,152 @@
+        ds_zero2 = DeepSpeedPlugin(
+            hf_ds_config=self.config_zero2,
+        )
+        ds_zero3 = DeepSpeedPlugin(
+            hf_ds_config=self.config_zero3 if not zero3_inference else self.config_zero3_inference,
+        )
+        return {"zero2": ds_zero2, "zero3": ds_zero3}
+
+    def test_enable_disable(self):
+        ds_plugins = self.get_ds_plugins()
+        ds_zero2, ds_zero3 = ds_plugins.values()
+        accelerator = Accelerator(
+            deepspeed_plugin=ds_plugins,
+        )
+        # Accelerator's constructor should automatically enable the first plugin
+        assert ds_zero2.enabled
+        assert not ds_zero3.enabled
+        assert get_active_deepspeed_plugin(accelerator.state) == ds_zero2
+        assert accelerator.deepspeed_plugin == ds_zero2
+        assert accelerator.state.get_deepspeed_plugin("zero2") == ds_zero2
+        accelerator.state.enable_deepspeed_plugin("zero3")
+        assert not ds_zero2.enabled
+        assert ds_zero3.enabled
+        assert get_active_deepspeed_plugin(accelerator.state) == ds_zero3
+        assert accelerator.deepspeed_plugin == ds_zero3
+        assert accelerator.state.get_deepspeed_plugin("zero3") == ds_zero3
+        accelerator.state.enable_deepspeed_plugin("zero2")
+        assert not ds_zero3.enabled
+        assert ds_zero2.enabled
+        assert get_active_deepspeed_plugin(accelerator.state) == ds_zero2
+        assert accelerator.deepspeed_plugin == ds_zero2
+        assert accelerator.state.get_deepspeed_plugin("zero2") == ds_zero2
+
+    def test_enable_disable_manually_set(self):
+        ds_plugins = self.get_ds_plugins()
+        ds_zero2, _ = ds_plugins.values()
+        with self.assertRaises(ValueError):
+            ds_zero2.enable()
+        accelerator = Accelerator(deepspeed_plugin=ds_plugins)
+        accelerator.state.enable_deepspeed_plugin("zero2")
+        with self.assertRaises(NotImplementedError):
+            ds_zero2.enabled = False
+        assert ds_zero2.enabled
+
+    def test_prepare_multiple_models_zero3_inference(self):
+        ds_plugins = self.get_ds_plugins(zero3_inference=True)
+        accelerator = Accelerator(deepspeed_plugin=ds_plugins)
+        # Using Zero-2 first
+        model1 = self.model_init()
+        optimizer = DummyOptim(model1.parameters())
+        scheduler = DummyScheduler(optimizer)
+
+        dataset = RegressionDataset()
+        dataloader = torch.utils.data.DataLoader(dataset, batch_size=1)
+        model1, optimizer, scheduler, dataloader = accelerator.prepare(model1, optimizer, scheduler, dataloader)
+        accelerator.state.enable_deepspeed_plugin("zero3")
+        model2 = self.model_init()
+        with self.assertLogs(level="WARNING") as captured:
+            model2 = accelerator.prepare(model2)
+            self.assertIn(
+                "A wrapped DeepSpeed engine reference is currently tied for this `Accelerator()` instance.",
+                captured.output[0],
+            )
+
```

Comment:
just flagging that this test is missing the crucial part of actually running training.

Reply:
Correct, that's for the later test/the script